### PR TITLE
Fix for compiling with clang on macOS 10.12.6

### PR DESCRIPTION
**Pull request by catull**
This PR fixes issue #264.
Now both clang and gcc build the sources.

### CODE REVIEW

File: src/lib/libast/include/ast_sys.h [hunk 1]
```diff
@@ -122,8 +122,10 @@ extern int	spawnvex_add(Spawnvex_t*, intmax_t, intmax_t, Spawnvex_f, void*);
 extern int	spawnvex_apply(Spawnvex_t*, int, int);
 extern intmax_t	spawnvex_get(Spawnvex_t*, int, int);
 extern int	spawnvex_close(Spawnvex_t*);
+#if !defined(__DARWIN_C_LEVEL) || !defined(__clang__)
```
Comment (krader1961) [inline src/lib/libast/include/ast_sys.h]: @catull, Can you explain this fix? It doesn't make sense to me. First, macOS 10.12 provides these two functions and defines `size_t`. And the declarations inside this conditional block are essentially the same as those provided by the system headers. The compiler diagnostic implies that `size_t` is either not defined or incorrectly defined.

Comment (catull) [reply src/lib/libast/include/ast_sys.h]: In short, because the functions are macro-expanded to
```c
extern size_t __builtin___strlcat_chk (char*, const char*, size_t, __builtin_object_size (char*, 2 > 1 ? 1 : 0));
extern size_t __builtin___strlcpy_chk (char*, const char*, size_t, __builtin_object_size (char*, 2 > 1 ? 1 : 0));
```

IMHO `size_t` is defined and is not the culprit here.
I have a few questions to you.

1. Can you compile the sources on macOS 10.13.x out of master ?
2. Why are those functions redefined if they "are essentially the same as those provided by the system headers" ?  What is the motivation ?

I have prepared a new patch, which also indents the preprocessor directives, because it is hard to follow the nesting of `#ifdef` / `#else` / `#endif`.
In the past, you could not have white-spaces before a `#` and after it, either.
Modern C compiler have supported it for a while now.
Not suggesting this is a style we should adopt, just saying the code is easier to follow.

**Comment by krader1961** [comment]
> Can you compile the sources on macOS 10.13.x out of master ?

Yes on my main server. But surprisingly, since it is a nearly identical config, building fails on my Macbook Pro. The only significant difference is the macOS patch level.

> What is the motivation ?

Hard to say with certainty but it appears these functions do not exist on Linux or AIX (and probably some other distros). The code doesn't use `strlcat()` AFAICT. Long term I'd like to see `strlcpy()` replaced with `strncpy()`. I'll double-check it in a few minutes but I think your more recent commit is probably the best compromise for dealing with the fact macOS and BSD distros provide these functions. However, it's a fragile fix because if any module includes the system *string.h* header before the AST header it will still break. But for now it's probably good enough.

**Comment by krader1961** [comment]
I've merged this. Thanks. But in the future please do not mix changes to the style of the code with changes to fix a specific issue.
